File: Assets/VivaFramework/Prayer/Common/DragHandler.cs
using UnityEngine;
using UnityEngine.EventSystems;

namespace Prayer
{
    public class DragHandler : OperateHandler, IBeginDragHandler, IDragHandler, IDropHandler, IEndDragHandler
    {
        
        public const string BEGIN_DRAG = "BeginDrag";
        public const string DRAG = "Drag";
        public const string END_DRAG = "EndDrag";
        public const string DROP = "Drop";
        
        public void OnBeginDrag(PointerEventData eventData)
        {
            TakeCall(BEGIN_DRAG, eventData);
        }

        public void OnDrag(PointerEventData eventData)
        {
            TakeCall(DRAG, eventData);
        }


        public void OnEndDrag(PointerEventData eventData)
        {
            TakeCall(END_DRAG, eventData);
        }

        public void OnDrop(PointerEventData eventData)
        {
            TakeCall(DROP, eventData);
        }
    }
    
}

File: LuaAPI/Prayer/DragHandler.lua
---@class Prayer.DragHandler : Prayer.OperateHandler
---@field BEGIN_DRAG string
---@field DRAG string
---@field END_DRAG string
---@field DROP string
local m = {}
---@param eventData UnityEngine.EventSystems.PointerEventData
function m:OnBeginDrag(eventData) end
---@param eventData UnityEngine.EventSystems.PointerEventData
function m:OnDrag(eventData) end
---@param eventData UnityEngine.EventSystems.PointerEventData
function m:OnEndDrag(eventData) end
---@param eventData UnityEngine.EventSystems.PointerEventData
function m:OnDrop(eventData) end
Prayer = {}
Prayer.DragHandler = m
return m